sdk/dotnet/Pulumi/Deployment/Deployment_Call.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Copyright 2016-2021, Pulumi Corporation
+
+using System;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Google.Protobuf.WellKnownTypes;
+using Pulumi.Serialization;
+using Pulumirpc;
+
+namespace Pulumi
+{
+    public sealed partial class Deployment
+    {
+        void IDeployment.Call(string token, CallArgs args, Resource? self, CallOptions? options)
+            => Call<object>(token, args, self, options, convertResult: false);
+
+        Output<T> IDeployment.Call<T>(string token, CallArgs args, Resource? self, CallOptions? options)
+            => Call<T>(token, args, self, options, convertResult: true);
+
+        private Output<T> Call<T>(
+            string token, CallArgs args, Resource? self, CallOptions? options, bool convertResult)
+        {
+            var completionSource = new OutputCompletionSource<T>(null);
```

Comment:
Haven't seen this one before. Is something tricky going on here?

What if we did use this constructor:

```
        internal Output(Task<OutputData<T>> dataTask) 
```

So ```

var task = CallAsync(....); // CallAsync now straight "returns" OutputData<_>
_runner.RegisterTask(.., task);
return Output.Create(task);  // this should take care of Exception handling ?

```